src/server/Main.cpp

```diff
@@ -129,9 +126,25 @@ int main() {
             [&server](size_t id, std::string name, std::vector<uint8_t> data) { server.update_component(id, name, data); }
         );
 
+        engine.registerComponent<Components::Visible>("./plugins/bin/components/", "Visible");
+        engine.registerComponent<Components::Health>("./plugins/bin/components/", "Health");
+        engine.registerComponent<Components::Collider>("./plugins/bin/components/", "Collider");
+        engine.registerComponent<Components::Scale>("./plugins/bin/components/", "Scale");
+
+        engine.loadSystems("./src/server/configServer.cfg");
+
+        server.updateIdStringToType(engine.getIdStringToType());
+
+        // engine.getRegistry().componentManager().
         while(server.client_endpoints.empty())
             server.start_receive();
 
+        std::thread io_thread([&io_context]() { io_context.run(); });
+        std::thread io_thread1([&io_context]() { io_context.run(); });
+        std::thread io_thread2([&io_context]() { io_context.run(); });
+        std::thread io_thread3([&io_context]() { io_context.run(); });
+        std::thread io_thread4([&io_context]() { io_context.run(); });
+
```

Comment:
I am not sure if this really help since all the operations that `io_context.run()` are already asynchronous.

Reply:
It says in the documentation of the run() function : 
``The run() function blocks until all work has finished and there are no more handlers to be dispatched, or until the io_context has been stopped.

Multiple threads may call the run() function to set up a pool of threads from which the io_context may execute handlers. All threads that are waiting in the pool are equivalent and the io_context may choose any one of them to invoke a handler.``

So i think its helping, tho, as of right now i dont think this implementation is usefull since the lags are actually in the client side.

A good thing to note is that we could create one thread per client using this format.

Reply:
Oh okay I see, very well